src/Arcus.Messaging.Pumps.ServiceBus/AzureServiceBusMessagePump.cs

```diff
@@ -437,20 +440,29 @@ private async Task HandleMessageAsync(Message message, CancellationToken cancell
                     Logger.LogInformation("No operation ID was found on the message");
                 }
 
+
                 MessageCorrelationInfo correlationInfo = message.GetCorrelationInfo();
-                Logger.LogInformation(
-                    "Received message '{MessageId}' (Transaction: {TransactionId}, Operation: {OperationId}, Cycle: {CycleId})",
-                     message.MessageId, correlationInfo.TransactionId, correlationInfo.OperationId, correlationInfo.CycleId);
-                
-                var messageContext = new AzureServiceBusMessageContext(message.MessageId, message.SystemProperties,
-                    message.UserProperties);
+                using (IServiceScope serviceScope = ServiceProvider.CreateScope())
+                {
+                    var correlationInfoAccessor = serviceScope.ServiceProvider.GetService<ICorrelationInfoAccessor<MessageCorrelationInfo>>();
```

Comment:
This would mean that the correlation accessor in observability should be registered as scoped.

Reply:
That's the case now so resolving, ok?

Reply:
Yes indeed